CDCgov/trusted-intermediary
# Eliminate warnings for missing fields

# Description

This PR updates and/or removes unnecessary fields in our FHIR mappings that were causing warnings when sending messages.

## Issue
#940 

When an extension contains a URL but no value of any type, the message "Resource is missing required element: value" is printed in the log.

There were 3 "value-less" extensions remaining that were producing these warnings.

1. Patient identifier type code `CX-5` (`PID-3.5` in the v2 message) - transform: `RemovePatientIdentifiers`
2. Patient name type code `XPN-7` (`PID-5.7` in the v2 message) - transform: `RemovePatientNameTypeCode`
3. Observation request `OBR-16` (occurs only when `ORC-12` is blank - transform: `CopyOrcOrderProviderToObrOrderProvider`

This PR ensures that the "value-less" extensions do not get added during the TI transforms and does some additional cleanup of the transform output to ensure that the values being cleared by the transforms no longer exist in other places in the affected resource.

See the [Areas of Concern comment in the issue](https://github.com/CDCgov/trusted-intermediary/issues/940#issuecomment-2518611083) for more details.

## Testing instructions

- Run a sample CA message with empty `ORC-12` and `OBR-16` through RS and TI end to end (such as CA examples 7, 19, or 20). Review the transformed fhir and final HL7.
- The transformed fhir should have no extensions without a value (specifically `CX.5`, `XPN.7`, and `OBR.16`)
- There should be no change in the final HL7



## Checklist
